include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld

```diff
@@ -266,36 +266,9 @@ SECTIONS
     _app_smem_rom_start = LOADADDR(_APP_SMEM_SECTION_NAME);
 #endif  /* CONFIG_USERSPACE */
 
-    SECTION_DATA_PROLOGUE(_BSS_SECTION_NAME,(NOLOAD), BSS_ALIGN)
-    {
-        /*
-         * For performance, BSS section is assumed to be 4 byte aligned and
-         * a multiple of 4 bytes
-         */
-        . = ALIGN(4);
-        __bss_start = .;
-        __kernel_ram_start = .;
-
-        *(.bss)
-        *(".bss.*")
-        *(COMMON)
-        *(".kernel_bss.*")
-
-#ifdef CONFIG_CODE_DATA_RELOCATION
-#include <linker_sram_bss_relocate.ld>
-#endif
-
-        /*
-         * As memory is cleared in words only, it is simpler to ensure the BSS
-         * section ends on a 4 byte boundary. This wastes a maximum of 3 bytes.
-         */
-        __bss_end = ALIGN(4);
-    } GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
-
-#include <zephyr/linker/common-noinit.ld>
-
     SECTION_DATA_PROLOGUE(_DATA_SECTION_NAME,,)
     {
+        __kernel_ram_start = .;
```

Comment:
The original `__kernel_ram_start = .;` under `.bss` which aligns to `BSS_ALIGN`. This is for the MPU or MMU region settings (MPU requires the address to be aligned to 32-byte or 64-byte, MMU requires 4k). I think you should also add a `_ALIGN` into `SECTION_DATA_PROLOGUE(_DATA_SECTION_NAME,,)`, since data section is the first one. Maybe rename the BSS_ALIGN to DATA_ALIGN or a more generic name, KERNEL_RAM_ALIGN?

Reply:
Anyway, __kernel_ram_start should align to `_region_min_align`